### DISCUSSION TITLE
Can a function be shared as a state?

### DISCUSSION THREAD

**Discussion by The-End-Hero** [facebookexperimental/Recoil#1270] [Q&A]
like this:
`export const changeItem = atom({
  key: "changeItem",
  default:  ()=>{},
});
`

**Comment by drarmstr** [comment]
Yes, you can store functions in atoms and selectors, but there are some nuances you may need to accomodate.

As the [`atom()`](https://recoiljs.org/docs/api-reference/core/atom) documentation says:

Atoms can be set to a function, as long as it is pure, but to do so you may need to use the updater form of setters. (e.g. `set(myAtom, () => myFunc);`).

For atom families you may also need to wrap in a function as well.

Also take care not to use functions in values that may need to be serialized to check for value-equality, such as atom family parameters.
